### ISSUE TITLE
Vivaldi browser completely broken specifically running under Hyprland (ozone issue)

### ISSUE THREAD

**Issue by mrshmllow** [hyprwm/Hyprland#216]
Vivaldi works completely fine on sway.

However, in hyprland, vivaldi does not start without `--ozone-platform=wayland`. Even then, it has a really hard time drawing, and will only update the screen if you unfocus vivaldi for a few seconds.
This occurs in both current `-stable` and `-snapshot` aur versions of vivaldi.

The same sort of draw lag occurs when i drag selected text in firefox

Disabling all decorations has no effect. Is there a remedy or option I dont know about?

**Comment by vaxerski** [comment]
chromium issue. For launching without ozone wayland, idk. Any logs?

**Comment by mrshmllow** [comment]
Chromium works fine.

Launched without ozone flag
```
[2695:2723:0614/175336.840624:ERROR:bus.cc(398)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[2695:2723:0614/175336.840662:ERROR:bus.cc(398)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[2695:2723:0614/175336.871383:ERROR:bus.cc(398)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[2695:2723:0614/175336.871395:ERROR:bus.cc(398)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[2695:2695:0614/175336.899168:ERROR:image_generic.cc(77)] Unable to decode PNG for 1.
libva error: vaGetDriverNameByIndex() failed with unknown libva error, driver_name = (null)
[2695:2695:0614/175337.937151:ERROR:CONSOLE(1)] "syncDetachedTabInformation: The message port closed before a response was received.", source: chrome-extension://mpognobbkildjkofajifpdfhcoklimli/bundle.js (1)
[2695:2695:0614/175337.937321:ERROR:CONSOLE(1)] "syncDetachedTabInformation: The message port closed before a response was received.", source: chrome-extension://mpognobbkildjkofajifpdfhcoklimli/bundle.js (1)
[2695:2695:0614/175337.937443:ERROR:CONSOLE(1)] "syncDetachedTabInformation: The message port closed before a response was received.", source: chrome-extension://mpognobbkildjkofajifpdfhcoklimli/bundle.js (1)
[2695:2754:0614/175338.331136:ERROR:bus.cc(398)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[2695:2695:0614/175338.331160:ERROR:object_proxy.cc(623)] Failed to call method: org.freedesktop.DBus.NameHasOwner: object_path= /org/freedesktop/DBus: unknown error type: 
[2695:2754:0614/175338.331194:ERROR:bus.cc(398)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[2695:2695:0614/175338.331227:ERROR:object_proxy.cc(623)] Failed to call method: org.freedesktop.DBus.NameHasOwner: object_path= /org/freedesktop/DBus: unknown error type: 
[2695:2754:0614/175338.331252:ERROR:bus.cc(398)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[2695:2695:0614/175338.331270:ERROR:object_proxy.cc(623)] Failed to call method: org.freedesktop.DBus.NameHasOwner: object_path= /org/freedesktop/DBus: unknown error type: 
[2695:2754:0614/175338.331297:ERROR:bus.cc(398)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[2695:2695:0614/175338.331307:ERROR:object_proxy.cc(623)] Failed to call method: org.freedesktop.DBus.NameHasOwner: object_path= /org/freedesktop/DBus: unknown error type: 
[2695:2754:0614/175338.331326:ERROR:bus.cc(398)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[2695:2695:0614/175338.331338:ERROR:object_proxy.cc(623)] Failed to call method: org.freedesktop.DBus.NameHasOwner: object_path= /org/freedesktop/DBus: unknown error type: 
[2695:2695:0614/175339.606156:ERROR:sharing_service.cc(222)] Device registration failed with fatal error
[2730:2730:0614/175341.118972:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 1 times!
[2695:2754:0614/175343.160291:ERROR:web_view_private_api.cpp(155)] No image from backing store.
[2730:2730:0614/175343.161221:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 2 times!
[2730:2730:0614/175343.161221:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 3 times!
```
It doesnt crash and runs headless after those

**Comment by vaxerski** [comment]
the stretching and shit on wayland ozone is a chromium issue. The "not launching in xwayland" is a vivaldi issue

**Comment by mrshmllow** [comment]
Weird. Works on sway.. I have no idea

**Comment by taylor85345** [comment]
> Weird. Works on sway.. I have no idea

I'm not sure why, but I've noticed that most Electron apps default to Ozone in other wlroots compositors like Sway and River, but it has to be specified manually in Hyprland.

**Comment by vaxerski** [comment]
FYI to all whom it may concern, parts of the issue have been resolved some time ago i've forgot to mention it, so I am working on the issue.

Positioning is fixed, however, ozone wayland doesn't seem to report damage properly, and thus is unable to update the view unless damage tracking is set to `none` (not recommended)

Wanted to let you know I'm working on the issue periodically and trying to find a fix.

**Comment by volucris1** [comment]
Chrome not working with ```--ozone-platform=wayland```:

https://user-images.githubusercontent.com/46374280/179203561-349f4c41-6962-4414-9891-6402285cbc69.mp4

**Comment by vaxerski** [comment]
yes we all know

**Comment by vaxerski** [comment]
fixed in 9be239dac63138282bac9e0acc4f91bb8e1f610b

**Comment by re-ynd** [comment]
Thank you for fixing the issue. Finally I was able to switch from sway to Hypr!
